### PR TITLE
Update mac x64 installer script arch detection

### PR DESCRIPTION
**Pull request by sfoslund**
Validated manually using both a x64 and arm64 machine

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Installers/src/GenerateMacOSDistributionFile.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public override bool Execute()
 function IsX64Machine() {
 var machine = system.sysctl(""hw.machine"");
 system.log(""Machine type: "" + machine);
-var result = machine == ""x64"" || machine.endsWith(""_x64"");
+var result = machine == ""x64"" || machine.endsWith(""_64"");
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Installers/src/GenerateMacOSDistributionFile.cs:47]: Can we make this match https://github.com/dotnet/runtime/pull/59210/files#diff-c6e8d33ed66dda3512254f78a84b3551527f3f67c4c128263ac130cc4d34f3e6R99
```suggestion
var result = machine == ""amd64"" || machine== ""x86_64"";
```

Comment (ericstj) [reply src/Microsoft.DotNet.Build.Tasks.Installers/src/GenerateMacOSDistributionFile.cs:47]: The original value here was based on me misreading a document that mentioned
`x86_64` (`x64`) .  I think that document was trying to communicate that it meant `x64` but was written as `x86_64` and I transposed the `x`.

File: src/Microsoft.DotNet.Build.Tasks.Installers/src/GenerateMacOSDistributionFile.cs [hunk 2]
```diff
@@ -42,11 +42,16 @@ public override bool Execute()
...
-system.log(""IsX64Machine: "" + result);
-return result;
+    var machine = system.sysctl(""hw.foo"");
+    var cputype = system.sysctl(""hw.cputype"");
+    var cpu64 = system.sysctl(""hw.cpu64bit_capable"");
+    system.log(""Machine type: "" + machine);
+    system.log(""Cpu type: "" + cputype);
+    system.log(""64-bit: "" + cpu64);
+    
+    var result = machine == ""amd64"" || machine == ""x86_64"" || cputype == ""16777223"" || (cputype == ""7"" && cpu64 == ""1"");
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Installers/src/GenerateMacOSDistributionFile.cs:52]: Let's add some comments around the magic numbers here.

// from machine.h
// CPU_TYPE_X86_64 = CPU_TYPE_X86 | CPU_ARCH_ABI64 = 0x010000007 = 16777223
// CPU_TYPE_X86 = 7

**Review by ericstj — APPROVED** [review]

File: src/Microsoft.DotNet.Build.Tasks.Installers/src/GenerateMacOSDistributionFile.cs [hunk 3]
```diff
@@ -42,11 +42,19 @@ public override bool Execute()
 
                 var archScriptContent = @"<![CDATA[
 function IsX64Machine() {
-var machine = system.sysctl(""hw.machine"");
-system.log(""Machine type: "" + machine);
-var result = machine == ""x64"" || machine.endsWith(""_x64"");
-system.log(""IsX64Machine: "" + result);
-return result;
+    var machine = system.sysctl(""hw.foo"");
```
Comment (am11) [inline src/Microsoft.DotNet.Build.Tasks.Installers/src/GenerateMacOSDistributionFile.cs:45]: is `.foo` a typo or is it to imply that `hw` `.anything` returns machine architecture?

**Pull request merged**